src/main/scala/gpp/ui/forms/FormInputEV.scala

```diff
@@ -72,14 +73,15 @@ object FormInputEV     {
   type ChangeCallback[A] = A => Callback
   type Backend           = RenderScope[Props, State, Unit]
 
-  final case class State(curValue: Option[String], changed: Boolean = false)
+  @Lenses
+  final case class State(curValue: Option[String], prevValue: String)
 
   def onTextChange(b: Backend): ReactEventFromInput => Callback =
     (e: ReactEventFromInput) => {
       // Capture the value outside setState, react reuses the events
       val v = e.target.value
       // First update the internal state, then call the outside listener
-      b.setState(State(v.some, changed = true)) *>
+      b.modState(State.curValue.set(v.some)) *>
```

Comment:
modStateL?

Reply:
Yes, good idea! Just pushed another commit with this changed.